lib/mayaUsd/ufe/UsdTransform3dPointInstance.cpp

```diff
@@ -0,0 +1,270 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "UsdTransform3dPointInstance.h"
+
+#include <mayaUsd/ufe/UsdPointInstanceOrientationModifier.h>
+#include <mayaUsd/ufe/UsdPointInstancePositionModifier.h>
+#include <mayaUsd/ufe/UsdPointInstanceScaleModifier.h>
+#include <mayaUsd/ufe/UsdPointInstanceUndoableCommands.h>
+#include <mayaUsd/ufe/UsdSceneItem.h>
+#include <mayaUsd/ufe/UsdTransform3dBase.h>
+#include <mayaUsd/ufe/UsdUndoableCommand.h>
+#include <mayaUsd/ufe/Utils.h>
+
+#include <pxr/base/gf/matrix4d.h>
+#include <pxr/base/gf/quatd.h>
+#include <pxr/base/gf/quath.h>
+#include <pxr/base/gf/rotation.h>
+#include <pxr/base/gf/transform.h>
+#include <pxr/base/gf/vec3d.h>
+#include <pxr/base/gf/vec3f.h>
+#include <pxr/base/vt/types.h>
+#include <pxr/usd/usd/attribute.h>
+#include <pxr/usd/usd/prim.h>
+#include <pxr/usd/usd/timeCode.h>
+
+#include <ufe/sceneItem.h>
+#include <ufe/transform3d.h>
+#include <ufe/transform3dHandler.h>
+#include <ufe/transform3dUndoableCommands.h>
+#include <ufe/types.h>
+
+#include <memory>
+
+PXR_NAMESPACE_USING_DIRECTIVE
+
+namespace {
+
+using namespace MayaUsd;
+using namespace MayaUsd::ufe;
+
+// Class for setMatrixCmd() implementation.
+class UsdPointInstanceSetMatrix4dUndoableCmd
```

Comment:
I know you're trying to implement the interface as completely as possible, but to my knowledge the setMatrixCmd() is called only by parent -absolute, and your instances cannot be reparented, can they?

I can't really think of another use case for this code --- you would need to have a workflow which would need an undoable command to set the instance's matrix, and currently I can't think of one; let me know if you think otherwise.

If that's the case, then this is essentially dead code --- and therefore, nice as it is :), we should probably take it out.

Reply:
Ahh, that explains some things actually, because I could figure out a way to get Maya to invoke this command. :)

That's true that it's not possible to re-parent point instances. I was thinking with this that maybe if you selected a point instance and then ran a `cmds.xform()` command to try to directly set the point instance's matrix that it might bottom out in this command, but that didn't seem to be the case.

I'll just remove this then if there's no way we can actually end up calling this command.

Reply:
Removed in rebased commit ac913871efe412ad175fa1f1adfc2850b7ff56f2.